smithy-model/src/main/java/software/amazon/smithy/model/knowledge/NeighborProviderIndex.java

```diff
@@ -76,6 +77,28 @@ public NeighborProvider getProviderWithTraitRelationships() {
         return result;
     }
 
+    /**
+     * Gets the neighbor provider that includes idRef relationships.
+     *
+     * @return Returns the provider.
+     */
+    public NeighborProvider getProviderWithIdRefRelationships() {
+        NeighborProvider result = providerWithIdRefs;
+
+        if (result == null) {
```

Comment:
I think this is one I wouldn't cache since this isn't used in selectors. I'd either just create it on demand, or entirely remove it from this index.

Reply:
Removed